code/game/machinery/doors/windowdoor.dm

```diff
@@ -103,18 +103,18 @@
 	icon_state = "[src.base_state]open"
 	flick("[src.base_state]opening", src)
 	playsound(src.loc, 'sound/machines/windowdoor.ogg', 100, 1)
-	addtimer(CALLBACK(src, .proc/open_final), 10, TIMER_UNIQUE | TIMER_OVERRIDE)
+	addtimer(CALLBACK(src, .proc/open_final), 0.9 SECONDS, TIMER_UNIQUE | TIMER_OVERRIDE)
 
 	return 1
 
-/obj/machinery/door/window/proc/open_final()
+/obj/machinery/door/window/open_final()
 	explosion_resistance = 0
 	set_density(FALSE)
 	update_icon()
 	update_nearby_tiles()
 
-	if(operating == 1) //emag again
-		operating = 0
+	if(operating) //emag again
```

Comment:
I believe the reason it says emag here is that emag sets `operating` to `-1` or some other value and this is meant to not un-emag the door. Please double-check, and if that's wrong and the `==1` is pointless here then update the comment.